File: container_mgmt/lib/models/yard.dart
class Yard {
  final int yardId;
  final int yardNumber;
  final int portId;
  final double? yardWidth;
  final double? yardHeight;
  final String? imagePath;

  Yard({
    required this.yardId,
    required this.yardNumber,
    required this.portId,
    this.yardWidth,
    this.yardHeight,
    this.imagePath,
  });

  bool get hasLayout => yardWidth != null && yardHeight != null;

  factory Yard.fromJson(Map<String, dynamic> json) => Yard(
    yardId: json['yardId'],
    yardNumber: json['yardNumber'],
    portId: json['portId'],
    yardWidth: (json['yardWidth'] as num?)?.toDouble(),
    yardHeight: (json['yardHeight'] as num?)?.toDouble(),
    imagePath: json['imagePath'] as String?,
  );
}

File: con_mgmt_api/Models/Yard.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ContainerManagementApi.Models;

[Table("Yards")]
public class Yard
{
    [Key]
    public int YardId { get; set; }

    [Required]
    public int YardNumber { get; set; }

    [Required]
    [Column("PortId")]
    public int PortId { get; set; }

    public double? YardWidth { get; set; }
    public double? YardHeight { get; set; }
}
